plugins/email_service/app/controllers/email_service/application_controller.rb

```diff
@@ -19,19 +19,58 @@ def restrict_access
         redirect_to index_path
       end
     end 
+
```

Comment:
wouldn't make more sence to do these checks with the policy? we used to use this way of checking roles because we wanted to redirec to to the old UI for testing purposes. All other roles check should go in the policy.

Reply:
Once we remove the role  `cloud_support_tools_viewer` in production, we need to check whether project has required roles and has credentials and `cronus` service is enabled or not. If one of them is missing it is supposed to render a page with hints and appropriate buttons to create credentials or enable cronus or add required roles. I think this check is required.

Reply:
The role checks are done by the policy engine if you have the policy configured correctly. You need to tell your controller `authorization_required` and then Elektra will show an error page stating the roles a user needs to be able to access the page without you having to do anything else. 

Also It is never a good idea to check for roles directly in a controller or view because you have to remember to change all the occurrences of the check if the role changes in any way.

Instead the correct check using the policy looks something like:
```
unless current_user.is_allowed?(email_service:email_list")
```

Reply:
One additional comment. In multi-layered boolean statements like the one below you should use parentheses to ensure the expression is evaluated the way you expect it to. So
```
unless (a && b) || (c && d)
```
And lastly, a double negative expression like the one above (starting with unless and then having a ! in the expression) is very hard to read and very easy to get wrong. I recommend simplifying it.

Reply:
sure. I have changed those double negation and inserted brackets for conditions. Removed custom role checks and let default behaviour do its work. Thanks.